lib/os/heap-validate.c

```diff
@@ -358,6 +358,8 @@ void heap_print_info(struct z_heap *h, bool dump_chunks)
 			allocated_bytes += chunksz_to_bytes(h, chunk_size(h, c));
 		} else if (!solo_free_header(h, c)) {
 			free_bytes += chunksz_to_bytes(h, chunk_size(h, c));
+		} else {
+			; /* Final statement (see coding guideline 15.7) */
```

Comment:
The comment shall describe the case why we end up here, not a self-describing comment ("final statement" in a final statement). I do not think we need to reference the guideline here as well.

Reply:
Yes, agreed that is the goal :) Here, to me it was not clear how to describe the condition based on context clues. So to avoid confusion, I elected to use an informative comment of why this non-empty else{} exists in the code. I hoped it could help make it harder to do the wrong thing in the future - like mistakenly remove this case and unknowingly contradict the CG. Code owners could instead improve upon this.

Can you please help suggest a more descriptive comment here? It would be great improvement. Otherwise, it is sufficient to do `;`. 

Also, thus far, we did not receive that feedback on the larger existing PR, so I'll incorporate this feedback in my next edits there as well. https://github.com/zephyrproject-rtos/zephyr/pull/33198

Reply:
ask @andyross  for the proper comment.

Reply:
I discussed with Andy and we are able to refactor this in a more intuitive way, considering that the chunk is either used or free, but not both. I've pushed the changes for review. Thanks!